### PR TITLE
add api for inference model

### PR DESCRIPTION
**Pull request by glorysdj**
this pr add simple pojo style inference model to analytics-zoo. 
java user can extends the AbstractInferenceModel and predict with java input and get java output

---
This change is [Reviewable](https://reviewable.io/reviews/intel-analytics/analytics-zoo/297)

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/FloatInferenceModel.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.intel.analytics.bigdl.nn.abstractnn.Activity
+import com.intel.analytics.bigdl.dataset.Sample
+import com.intel.analytics.bigdl.tensor.Tensor
+import scala.collection.JavaConverters._
+
+case class FloatInferenceModel(
+  model: AbstractModule[Activity, Activity, Float],
+  predictor: LocalPredictor[Float]) extends InferenceSupportive {
+
+  def forward(data: Tensor[Float])(implicit om: Manifest[Float]): Array[Float] = {
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/FloatInferenceModel.scala]: do we need this?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/FloatInferenceModel.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    model.forward(data).asInstanceOf[Tensor[Float]].storage().array()
+  }
+
+  def predict(samples: Array[Sample[Float]]): Array[Int] = {
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/FloatInferenceModel.scala]: do we need this?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/FloatInferenceModel.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    predictor.predictClass(samples)
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/FloatInferenceModel.scala]: why predictClass?

**Comment by jason-dai** [comment]
Is this thread safe?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/FloatInferenceModel.scala (hunk 1 above)
Comment (glorysdj) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/FloatInferenceModel.scala]: for scala user?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/FloatInferenceModel.scala (hunk 3 above)
Comment (glorysdj) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/FloatInferenceModel.scala]: I will delete these two functions for now. we can add more apis for both java and scala users in the future., imo.

**Comment by glorysdj** [comment]
For thread safe, @zhichao-li please help to review transferInferenceInputToTensor method which is used to transfer java List to bigdl Tensor. I'm not sure of this part.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceModelFactory.scala [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+object InferenceModelFactory{
+  def loadFloatInferenceModel(modelPath: String): FloatInferenceModel = {
+    loadFloatInferenceModel(modelPath, null)
+  }
+
+  def loadFloatInferenceModel(modelPath: String, weightPath: String): FloatInferenceModel = {
+    val model = ModelLoader.loadFloatModel(modelPath, weightPath)
+    val predictor = LocalPredictor(model)
+    model.evaluate()
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceModelFactory.scala]: remove this line ? we should suppose `LocalPredictor` has already done such job.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceSupportive.scala [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    val begin = System.currentTimeMillis
+    val result = f
+    val end = System.currentTimeMillis
+    val cost = (end - begin)
+    logger.info(s"$name time elapsed [${cost / 1000} s, ${cost % 1000} ms].")
+    result
+  }
+
+  def transferInferenceInputToTensor(input: java.util.List[java.util.List[java.lang.Float]]):
+    Tensor[Float] = {
```
Comment (zhichao-li) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceSupportive.scala]: should we add another parameter `shape` here?
and also we can simplified the logic like this:
1) convert input to an `Array` , you can reuse the logic below
```
while (d < arrays.length) {
+          System.arraycopy(arrays(d), 0, _buffer.storage().array(), d * size, size)
+          d += 1
+        }
2. create a tensor with `Tensor(data=xxArray, shape=xxshape)`
```

Comment (glorysdj) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/inference/InferenceSupportive.scala]: sure, we should add such api in the future.

**Pull request merged**